### PR TITLE
Add elasticsearch security settings

### PR DESCRIPTION
**Pull request by Jotschi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
 		<asm.version>3.3.1</asm.version>
 		<spring.security.version>4.2.12.RELEASE</spring.security.version>
 		<ferma.version>${project.version}</ferma.version>
-		<elasticsearch.client.version>0.5.2</elasticsearch.client.version>
+		<elasticsearch.client.version>1.0.0-SNAPSHOT</elasticsearch.client.version>
```
Comment (Jotschi) [inline pom.xml]: Use 1.0.0 when ready

File: api/src/main/java/com/gentics/mesh/etc/config/search/ElasticSearchOptions.java [hunk 2]
```diff
@@ -44,12 +51,38 @@
...
+	private String certPath;
+
+	@JsonProperty(required = false)
+	@JsonPropertyDescription("Path to the trusted common authority certificate (PEM format)")
+	@EnvironmentVariable(name = MESH_ELASTICSEARCH_CA_PATH_ENV, description = "Override the configured common authority certificate path.")
+	private String caPath;
+
+	@JsonProperty(required = false)
+	@JsonPropertyDescription("Flag which controls whether hostname verification should be enabled. Default: " + DEFAULT_HOSTNAME_VERIFICATION)
+	@EnvironmentVariable(name = MESH_ELASTICSEARCH_HOSTNAME_VERIFICATION_ENV, description = "Override the configured elasticsearch password.")
```
Comment (Jotschi) [inline api/src/main/java/com/gentics/mesh/etc/config/search/ElasticSearchOptions.java]: Fix desc

File: doc/src/main/docs/elasticsearch.asciidoc [hunk 3]
```diff
@@ -98,7 +98,55 @@ NOTE: The Gentics Mesh UI currently requires the Elasticsearch support to functi
 
 == Compatibility
 
-We currently run and test against Elasticsearch version 6.1.2. Other versions have not yet been tested.
+We currently run and test against Elasticsearch version 6.8.0. Other versions have not yet been tested.
+
+== Security
+
+Gentics Mesh support basic authentication and TLS in-transit encrypted connections for Elasticsearch communication.
```
Comment (philippguertler) [inline doc/src/main/docs/elasticsearch.asciidoc]: supports

File: doc/src/main/docs/elasticsearch.asciidoc [hunk 4]
```diff
@@ -98,7 +98,55 @@ NOTE: The Gentics Mesh UI currently requires the Elasticsearch support to functi
...
+  url: "https://127.0.0.1:9200"
+  username: "elastic"
+  password: "iucee7dohjaedemiShoshie9eiz4af0Oiceish6a"
+  certPath: "certs/elastic-certificates.crt.pem"
+  caPath: "certs/elastic-stack-ca.crt.pem"
+  hostnameVerification: true
+  startEmbedded: false
+```
+
+NOTE: The `certPath` and `caPath` setting only accept certificate files in `PEM` format.
```
Comment (philippguertler) [inline doc/src/main/docs/elasticsearch.asciidoc]: settings

File: doc/src/main/docs/elasticsearch.asciidoc [hunk 5]
```diff
@@ -98,7 +98,55 @@ NOTE: The Gentics Mesh UI currently requires the Elasticsearch support to functi
...
+
+NOTE: Elasticsearch servers which already use a trusted certificate don't need to specify this cert in the configuration. It can however be configured to only trust the given cert chain.
+
+The settings can be also configured via environment variables:
+
+* `MESH_ELASTICSEARCH_CERT_PATH` - Override the cert path
```
Comment (philippguertler) [inline doc/src/main/docs/elasticsearch.asciidoc]: Use table with types and default values

**Comment by Jotschi** [comment]
buildnow

**Pull request merged**